ledokku-bootstrap.sh

```diff
@@ -1,39 +1,162 @@
 #!/bin/bash
-# This script will install ledokku on your server
-# You need to already have dokku installed in order to be able to run it
+
+# This script is used to configure the essentials of t2d.
+## 1 => Check whether the program/application, "Whiptails" exists or not.
+## 2 => Making sure that the script is runing with root permissions.
+## 3 => Update and Upgrade the VPS.
+## 4 => Check whether the program/application, "Dokku" was Installed or not.
+## 5 => Upgrading dokku to the latest version.
+## 6 => Check Whether Plugins are installed or not (Redis, Postgres)
+## 7 => Install Redis and Postgress, if they were not installed
+## 7 => Install Ledokku
+## ? => Update Script (Will be added later)
+
 set -e
 
-# Check that dokku is installed on the server
-ensure-dokku() {
-  if ! command -v dokku &> /dev/null
-  then
-      echo "dokku is not installed, please follow our getting started guide first"
-      echo "https://www.ledokku.com/docs/getting-started"
+function define-colors {
+
+  # Using tput will eliminate the usage of "-e" in echo, and can be used anywhere
+  ## Color Palet
+  ## Should exist in every script
+  RED="$(tput setaf 1)" # ${RED}
+  GREEN="$(tput setaf 2)" # ${GREEN}
+  YELLOW="$(tput setaf 3)" # ${YELLOW}
+  BLUE="$(tput setaf 123)" # ${BLUE}
+  END="$(tput setaf 7)" # ${END
+}
+
+function system-info() {
+  
+  # Finding Information about your device
+  ## Basic VPS info
+  ## Should exist in every script
+  IP="$(ifconfig | grep broadcast | awk '{print $2}')"
```

Comment:
is this used? We already get the ip line 166

Reply:
If there are more than 2 IP addresses, it can be useful, or else we can remove this.

Reply:
We were using ifconfig but sometimes it reported the internal ip and using the approach line 166 was proven more stable across all installs so it's better to keep it that way :)

Reply:
**Edit 01**: So, I will comment out this, in the next commit before we can merge.

anything else?